plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/VspResourceAnswer.java

```diff
@@ -39,4 +39,22 @@ public VspResourceAnswer(VspResourceCommand cmd, Exception e) {
     public String getResourceInfo() {
         return this._resourceInfo;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        VspResourceAnswer that = (VspResourceAnswer) o;
+
+        if (_resourceInfo != null ? !_resourceInfo.equals(that._resourceInfo) : that._resourceInfo != null)
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return _resourceInfo != null ? _resourceInfo.hashCode() : 0;
+    }
```

Comment:
Please implement ``toString()`` to provide debugging state information for debugging purposes.